### PR TITLE
Csl 2059 net sdk add support for quizzes api

### PR DESCRIPTION
**Pull request by mocca102**

### CODE REVIEW

File: src/constructor.io/models/Quizzes/QuizRequest.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+        /// <summary>
+        /// Gets or sets version identifier for the quiz.
+        /// </summary>
+        public string VersionId { get; set; }
+
+        /// <summary>
+        /// Gets or sets user test cells.
+        /// </summary>
+        public Dictionary<string, string> TestCells { get; set; }
```
Comment (stanlp1) [inline src/constructor.io/models/Quizzes/QuizRequest.cs:37]: Not sure if we should include section/test cells here.

For section, I don't think there's currently a way to create quizzes for other sections.

For test cells, I don't think the api actually does anything with it.

**Review by stanlp1 — CHANGES_REQUESTED** [review]
Small comments. Otherwise it looks good to me.

File: src/Constructorio_NET.Tests/client/modules/QuizzesTests.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+using System.IO;
+using System.Threading.Tasks;
+using Constructorio_NET.Models;
+using Newtonsoft.Json.Linq;
+using NUnit.Framework;
+
+namespace Constructorio_NET.Tests
+{
+    [TestFixture]
+    public class QuizzesTest
```
Comment (stanlp1) [inline src/Constructorio_NET.Tests/client/modules/QuizzesTests.cs:12]: Can we add a test that checks if all the attributes in a quiz question are there, correct? Like the description, id, title, images...

That way we can be sure that the attributes deserialized correctly.

File: src/constructor.io/utils/Helpers.cs [hunk 3]
```diff
@@ -163,6 +163,18 @@ protected static string MakeUrl(Hashtable options, List<string> paths, Hashtable
                     }
                 }
 
+                // Add quiz answers to query string
+                if (queryParams.Contains(Constants.ANSWERS))
```
Comment (stanlp1) [inline src/constructor.io/utils/Helpers.cs:167]: Nice catch, missed this one.

Looking good to me. Lets have someone else have a final pass.

**Review by stanlp1 — APPROVED** [review]
LGTM.

File: src/constructor.io/models/Quizzes/QuestionOption.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    {
+        [JsonProperty("id")]
+        public int Id { get; set; }
+
+        [JsonProperty("value")]
+        public string Value { get; set; }
+
+        [JsonProperty("attribute")]
+        public OptionAttribute Attribute { get; set; }
+    }
```
Comment (Mudaafi) [inline src/constructor.io/models/Quizzes/QuestionOption.cs:20]: I believe the `multiple` and `single` question types have an [image property in options](https://docs.constructor.io/rest_api/quiz/using_quizzes/#multiple-select).

**Review by Mudaafi — APPROVED** [review]
LGTM! Thanks for making the changes <3

**Pull request merged**